test/e2e/serial/tests/resource_accounting.go

```diff
@@ -261,21 +277,37 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload resourc
 			schedOK, err := nrosched.CheckPODWasScheduledWith(fxt.K8sClient, updatedPod.Namespace, updatedPod.Name, serialconfig.Config.SchedulerName)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod.Namespace, updatedPod.Name, serialconfig.Config.SchedulerName)
+
+			//Check that NRT of the target node reflect correct consumed resources
+			By("Verifying NRT is updated properly when running the test's pod")
+			targetNrtListAfter, err := e2enrt.GetUpdated(fxt.Client, targetNrtListBefore, 1*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+			targetNrtAfter, err := e2enrt.FindFromList(targetNrtListAfter.Items, targetNodeName)
+			Expect(err).NotTo(HaveOccurred())
+
+			dataBefore, err := yaml.Marshal(targetNrtBefore)
+			Expect(err).ToNot(HaveOccurred())
+			dataAfter, err := yaml.Marshal(targetNrtAfter)
+			Expect(err).ToNot(HaveOccurred())
+			match, err := e2enrt.CheckZoneConsumedResourcesAtLeast(*targetNrtBefore, *targetNrtAfter, requiredRes)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(match).ToNot(Equal(""), "inconsistent accounting: no resources consumed by the running pod,\nNRT before test's pod: %s \nNRT after: %s \npod resources: %v", dataBefore, dataAfter, e2ereslist.ToString(requiredRes))
```

Comment:
better: `Expect(match).ToNot(BeEmpty())` (here and below)

Reply:
Additionally, lines 288-294 seems a good candidate for a test helper

Reply:
actually no. This needs to be re-evaluated carefully, perhaps rethinking `CheckZoneConsumedResourcesAtLeast` along the way